packages/manager/apps/catalog-revamp/src/pages/index.tsx

```diff
@@ -1,35 +1,72 @@
 import React from 'react';
 import { useTranslation } from 'react-i18next';
-import { useQuery, UseQueryResult } from '@tanstack/react-query';
 import {
-  getManagerHubCatalogList,
-  getManagerHubCatalogListQueryKey,
-} from '@/api';
-import Error from '@/components/Error/Error';
-
-interface ServiceData {
-  status: number;
-  data: any;
-}
+  ODS_THEME_COLOR_INTENT,
+  ODS_THEME_SIZE,
+  ODS_THEME_TYPOGRAPHY_LEVEL,
+} from '@ovhcloud/ods-common-theming';
+import { OsdsText } from '@ovhcloud/ods-components/text/react/';
+import { OsdsDivider } from '@ovhcloud/ods-components/divider/react/';
+import { OsdsSkeleton } from '@ovhcloud/ods-components/skeleton/react';
+import { MscTile } from '@ovhcloud/msc-react-tile';
+import { useCatalog } from '@/hooks/useCatalog';
+import SearchBar from '@/components/SearchBar/SearchBar';
+import { Product } from '@/utils/utils';
+import Loading from '../components/Loading/Loading';
 
 export default function CatalogRevamp() {
   const { t } = useTranslation('catalog-revamp');
+  const [searchText, setSearchValue] = React.useState('');
+  const [selectedCategories, setSelectedCategories] = React.useState<string[]>(
+    [],
+  );
+  const [selectedUniverses, setSelectedUniverses] = React.useState<string[]>(
+    [],
+  );
 
-  const service: UseQueryResult<ServiceData> = useQuery({
-    queryKey: getManagerHubCatalogListQueryKey,
-    queryFn: () => getManagerHubCatalogList(),
-    staleTime: Infinity,
+  const { results, products, isLoading } = useCatalog({
+    categories: selectedCategories,
+    universes: selectedUniverses,
+    searchText,
   });
 
-  if (!service.isLoading && service.data.status !== 200) {
-    return <Error error={service.data} />;
-  }
-  const data = service.data?.data?.catalog?.data;
+  const getResultsNumber = () => {
+    return ` (${results.length})`;
+  };
+
   return (
-    <div>
-      <h1>{t('title')}</h1>
-      <div>Start your application</div>
-      <div>{JSON.stringify(data)}</div>
+    <div className="m-10">
+      <OsdsText
+        level={ODS_THEME_TYPOGRAPHY_LEVEL.heading}
+        size={ODS_THEME_SIZE._600}
+        color={ODS_THEME_COLOR_INTENT.text}
+      >
+        {t('title') + getResultsNumber()}
+      </OsdsText>
+      <SearchBar
+        products={products}
+        setSelectedCategories={setSelectedCategories}
+        setSelectedUniverses={setSelectedUniverses}
+        setSearchValue={setSearchValue}
+      />
+      <OsdsDivider separator />
+      <div className="grid grid-cols-1 md:grid-cols-2 lg:grid-cols-3 gap-x-5 gap-y-5 pt-3">
+        {results.map((item: Product, id: number) => (
+          <div
+            key={id}
+            className="break-words p-4 border-2 border-solid border-cyan-200 rounded-xl"
+          >
+            <MscTile
+              key={item.id}
+              tileTitle={item.name}
+              tileType={'product'}
```

Comment:
```suggestion
              category={item.category}
```

or something like that